torchtext/experimental/transforms.py

```diff
@@ -144,22 +147,169 @@ class TextSequentialTransforms(nn.Sequential):
+        Args:
+            tokens: the tokens list for decoder
+
+        Examples:
+            >>> spm_transform.decoder(['▁the', '▁pre', 'trained', '▁sp', '▁model', '▁names'])
+            >>> 'the pretrained sp model names'
+        """
+
+        return self.sp_model.DecodePieces(tokens)
+
+
+class PretrainedSPTransform(nn.Module):
+    r"""string to ids transform based on a pretained sentencepiece model
+
+    Args:
+       spm_model: the pretrained sentencepiece model names. Default: 'text_unigram_25000'. The following pretrained sentencepiece models are provided:
+            - text_unigram_15000
+            - text_unigram_25000
+            - text_unigram_50000
+            - text_bpe_15000
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import PretrainedSPTransform
+        >>> spm_transform = PretrainedSPTransform('text_unigram_25000')
+        >>> spm_transform = PretrainedSPTransform('.data/text_unigram_25000.model')
+        >>> jit_spm_transform = torch.jit.script(spm_transform)
+    """
+
+    def __init__(self, spm_model='text_unigram_25000'):
+        super(PretrainedSPTransform, self).__init__()
+        if spm_model in _pretrained_spm:
+            spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        try:
+            self.sp_model = load_sp_model(spm_model)
+        except:
+            raise RuntimeError('The pretrained sentencepiece model is not supported')
+
+    def forward(self, line: str) -> List[int]:
+        r"""
+        Args:
+            line: the input string
+
+        Examples:
+            >>> spm_transform('the pretrained sp model names')
+            >>> [9, 1546, 18811, 2849, 2759, 2202]
+        """
+
+        return self.sp_model.EncodeAsIds(line)
+
+    @torch.jit.export
+    def decode(self, ids: List[int]) -> str:
+        r"""
+        Args:
+            ids: the integer list for decoder
+
+        Examples:
+            >>> spm_transform.decoder([9, 1546, 18811, 2849, 2759, 2202])
+            >>> 'the pretrained sp model names'
+        """
+
+        return self.sp_model.DecodeIds(ids)
+
+
+class ToLongTensor(nn.Module):
+    r"""Convert a list of integers to long tensor
+
+    Examples:
+        >>> from torchtext.experimental.transforms import ToLongTensor
+        >>> to_tensor = ToLongTensor()
+    """
+
+    def __init__(self):
+        super(ToLongTensor, self).__init__()
+
+    def forward(self, ids: List[int]) -> Tensor:
+        r"""
+        Args:
+            ids: the input list of ids
+
+        Examples:
+            >>> to_tensor([9, 1546, 18811, 2849, 61, 2759, 2202])
+            >>> tensor([    9,  1546, 18811,  2849,    61,  2759,  2202])
+        """
+        return torch.tensor(ids).to(torch.long)
```

Comment:
The torch tensor constructor has a dtype argument if you want to avoid this additional copy. You can do

torch.tensor(ids, dtype=torch.long).

But you're probably better served to use torch.as_tensor, which can deal with all kinds of data like numpy, list, tuple, scalar, ndarray etc.: https://pytorch.org/docs/master/generated/torch.as_tensor.html#torch.as_tensor

I understand that this wrapper is really just needed for JIT support?

Reply:
Yes, it's for the JIT support. And also people have a one-stop transform to do tokenization + vocab.